### DISCUSSION TITLE
defining the BC regarding the defromation gradient

### DISCUSSION THREAD

**Discussion by mFathidoost** [idaholab/moose#19487] [Q&A Modules: Solid mechanics] [answered]
HI,

I have a given gradient defromatin matrix. I need to define the BC of a 2D RVE in order to deform corresponding to the given gradient deformation. Do you know if MOOSE is possible to do that for me automatically? It mean that instead of BC I just enter the deformation gradient matrix. If yes via which module?

thank you very much

**Comment by hugary1995** [comment, marked as answer]
I suppose by deformation gradient you are implying large deformation. You can use the global strain system in the tensor mechanics module to do that. 

The global strain system isn't strictly equivalent to a standard KUBC. There's a new PR #17475 to address that.
